testsuite/MDAnalysisTests/lib/test_util.py

```diff
@@ -605,103 +598,142 @@ class TestGuessFormat(object):
         ('XYZ', mda.topology.XYZParser.XYZParser, mda.coordinates.XYZ.XYZReader),
     ]
     if six.PY2:
-        # DCD, LAMMPS, and TRZ are not supported on Python 3 yet
+        # TRZ is not supported on Python 3 yet
         formats += [
-            ('DATA', mda.topology.LAMMPSParser.DATAParser,
-             mda.coordinates.LAMMPS.DATAReader),
-            ('DCD', None, mda.coordinates.DCD.DCDReader),
-            ('LAMMPS', None, mda.coordinates.LAMMPS.DCDReader),
             ('TRZ', None, mda.coordinates.TRZ.TRZReader),
         ]
     # list of possible compressed extensions
     # include no extension too!
     compressed_extensions = ['.bz2', '.gz']
 
-    def _check_get_ext(self, f, fn):
+    @pytest.mark.parametrize('extention',
+        map(lambda format: format[0].upper(), formats) +
+        map(lambda format: format[0].lower(), formats)
+    )
+    def test_get_extention(self, extention):
         """Check that get_ext works"""
-        a, b = util.get_ext(fn)
+        file_name = 'file.{0}'.format(extention)
+        a, b = util.get_ext(file_name)
 
-        assert_equal(a, 'file')
-        assert_equal(b, f.lower())
+        assert a == 'file'
+        assert b == extention.lower()
 
-    def _check_compressed(self, f, fn):
+    @pytest.mark.parametrize('extention',
+        map(lambda format: format[0].upper(), formats) +
```

Comment:
turn it into a list comprehension `[f[0].upper() for f in formats]`. Please also try not to use keywords of the language itself.